include/IECorePython/VectorTypedDataBinding.inl

```diff
@@ -513,7 +535,7 @@ class VectorTypedDataFunctions
 		{
 			return boost::python::object(
 				boost::python::handle<>(
-					PyString_FromStringAndSize( reinterpret_cast<const char*>( x.baseReadable() ), x.baseSize() * sizeof( typename ThisClass::BaseType ) )
+					PyBytes_FromStringAndSize( reinterpret_cast<const char*>( x.baseReadable() ), x.baseSize() * sizeof( typename ThisClass::BaseType ) )
```

Comment:
I've found this reference https://py3c.readthedocs.io/en/latest/reference.html when lookin into this while trying to compile cortex with python3 on my machine.

So I wonder if `PyString_FromStringAndSize` should actually be replaced by `PyUnicode_FromStringAndSize` rather than `PyBytes_FromStringAndSize`. Unless the value it handles is bytes?

I'm not an expert :) ... I'm hoping you know better and I will learn something too.

Reply:
Rough recap : `str` in Python 2 is equivalent to `bytes` in Python 3, and `unicode` in Python 2 is equivalent to `str` in Python 3. In Python 2, we often used `str` to store/manipulate text when we should have been using `unicode`. One of Python 3's appeals is that it forces us into a more principled approach, where we use the new `str` exclusively for text, and are forced to encode to/from `bytes` at appropriate boundaries.

Given that explanation, and the naming of `toString()`, you could be forgiven for thinking that `PyUnicode_FromStringAndSize()` is the right approach here, but it's not. Despite the name, `toString()` was intended to expose the underlying data in raw form, and hence `toBytes()` would have been a more suitable name all along. In my first pass, I actually ripped it out completely on the grounds that it was ill conceived, confusing and unused. But then I discovered one use of it (in Gaffer) that I couldn't easily avoid, so it got a stay of execution. In that use, we are definitely expecting raw data, and I've tested that the `bytes` approach is correct.
 
Arguably we should be supporting the [buffer protocol](https://docs.python.org/3/c-api/buffer.html) as a superior replacement for `toString()`. But the scope of the Python 3 work is huge, so I'm trying to make pragmatic choices about how much I improve things and how much I just port as-is.

Reply:
Yes, it was a bit confusing for me. However your explanation puts everything in place. Buffers are looking interesting too. I suppose it could make it easier to manipulating big data structures, or to tweak them.

Anyway, it's nice to see that Python 3 support is coming fast.
Thanks John.